students/Matt_Hudgins/Lesson09/mailroom_5.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+# chmod +x mailroom_5.py needs to be performed before executable
+
+'''
+    File Name: mailroom_5.py
+    Author: Matt Hudgins
+    Date created: 6/3/18
+    Date last modified: 6/3/18
+    Python Version 3.6.4
+'''
+import sys
+
+#Need Donor Class
+#Need Donor Collection Class
+#Interaction code should be outside the "model" class
+
+class Donor:
+
+    def __init__(self, name, donation=None):
+        self.donation = donation
+        self.name = name
+        self.total = sum(self.donation)
+        if donation is None:
+            self.donation = []
+        else:
+            self.donation = donation
+
+    @property
+    def add_donation(self, amount):
+        self.donation.append(amount)
+        self.total += amount
+        self.average = self.total / len(self.donation)
+
+class DonorDatabase:
+
+    def __init__(self, name=None):
+        self.donors = [] if name is None else name
+
+
+    def add_donor(self, name):
+        self.donors.append(name)
+
+
+    def get_donor(self, name):
+        for d in self.donors:
+            if d.name == name:
+                return d
+
+    def get_all_donor_names(self):
+        return [donor.name for donor in self.donors]
+
+
+    def sort_donors(self):
+        return sorted(self.get_all_donor_names())
+
+
+    def donor_input(self):
+        return input("Enter a donor name or input 'List' for a list of donors\n>")
+
+
+    def send_thankyou(self):
+        don_input = None
+        while not don_input:
+            don_input = self.donor_input()
```

Comment:
keep in mind the separation of concern here; you should have no input functionality in your Data class.

if you were to "upgrade" your mailroom to have UI instead of CLI, there should be no changes to `Donor` or `DonorDatabase` classes.

as it is now you will have to make some changes.